[INF-0002] fix comment about resource not fully bound. (#238)
microsoft/hlsl-specs: proposals/infra/INF-0002-root-signature-in-clang.md

Most values like ShaderVisibility/ParameterType are covered by syntactical 
checks in Sema.

> The additional sematic rules not already covered by the grammar are listed here.

Reviewer: ```suggestion
The additional semantic rules not already covered by the grammar are listed here.
```

#### Resource used in DXIL must be fully bound in root signature.
```

>   // B is bound to t1, but not in root signature.

Reviewer: ```suggestion
  // B is bound to t1, but no root parameters cover t1.
```

Reviewer: The root signature says nothing about resources, it tells you about registers.
